tests/integration/records/single_db/test_records_unload_datetime.py

```diff
@@ -0,0 +1,364 @@
+                                           hints={
+                                               'datetimeformattz': datetimeformattz,
+                                               'compression': None,
+                                               'header-row': False,
+                                               **addl_hints,  # type: ignore
+                                           })
+            expect_pandas_failure = (not self.has_pandas()) and uses_pandas
+            try:
+                csv_text = self.unload_column_to_string(column_name='timestamptz',
+                                                        records_format=records_format)
+                self.assertIn(csv_text, [create_sample(datetimeformattz) + "\n",
+                                         create_sample(datetimeformattz).
+                                         replace('-00', '+00') + "\n",
+                                         create_sample(datetimeformattz).replace('-00', '') +
+                                         ".000000\n",
+                                         # Pandas doesn't truncate fractional seconds in the
+                                         # same way other tools do.
+                                         create_sample(datetimeformattz) + ".000000\n",
+                                         # It's valid to not include an offset with no impact.
+                                         create_sample(datetimeformattz).replace('-00', '') +
+                                         '.000000+0000\n',
+                                         # An example of having an overly simplistic
+                                         # implementation of to_csv_options is that seconds get
+                                         # appended even for time formats which don't include
+                                         # it.
+                                         #
+                                         # https://github.com/bluelabsio/records-mover/issues/142
+                                         create_sample(datetimeformattz) +
+                                         f":{SAMPLE_SECOND:02d}.000000\n"
+                                         ],
+                              f"from datetimeformattz {datetimeformattz} and "
+                              f"addl_hints {addl_hints}")
+            except ModuleNotFoundError as e:
+                if 'pandas' in str(e) and expect_pandas_failure:
+                    # as expected
+                    continue
+                else:
+                    raise
+            except NotImplementedError as e:
+                if expect_am_failure and 'AM' in str(e):
+                    # as expected
+                    continue
+                else:
+                    raise
+            self.assertFalse(expect_pandas_failure, datetimeformattz)
+            self.assertFalse(expect_am_failure, datetimeformattz)
+
+    def test_unload_timeonly(self) -> None:
```

Comment:
C901 'RecordsUnloadDatetimeIntegrationTest.test_unload_timeonly' is too complex (17)